orderer/consensus/etcdraft/storage.go

```diff
@@ -441,6 +443,7 @@ func (rs *RaftStorage) ApplySnapshot(snap raftpb.Snapshot) {
 
 // Close closes storage
 func (rs *RaftStorage) Close() error {
+	close(rs.WALSyncC)
```

Comment:
why do you close this channel? you are writing to it in line 1153 in `chain.go` isn't that a problem?

Reply:
The channel is getting closed to release the resource when the chain is halted.

Reply:
To release what resource?

If the chain is still referenced by a goroutine then closing it won't do any good. If it's not referenced by any goroutine then the channel will also be garbage collected. 

But I am asking - what if the chain goroutine will write to the closed channel after it's closed?

Reply:
And seems like the unit test failed on the exact send:

```

==================
panic: send on closed channel

goroutine 2985 [running]:
github.com/hyperledger/fabric/orderer/consensus/etcdraft.(*Chain).apply(0xc007304300, {0xc0072b9830, 0x1, 0x0?})
	/home/runner/work/fabric/fabric/orderer/consensus/etcdraft/chain.go:1153 +0x52b
```